### DISCUSSION TITLE
Can span be disabled after created?

### DISCUSSION THREAD

**Discussion by MrCroxx** [tokio-rs/tracing#3014] [Q&A]
Hi, I tried to use `tracing` to do tail-based tracing, which requires to disable a span after it is created or even entered. Is that possible now?

**Comment by mladedav** [comment]
What does tail-based tracing mean? And what do you mean by disabling a span, filtering it out the next time it should be created?

**Reply by MrCroxx** [reply]
Hi @mladedav . By tail-based tracing, I mean to cancel a span tree based on the total duration of the root span. I'm not sure if it can be achieved with low overhead with *tracing*. With *minitrace*, there is a `cancel()` method for `Span` that can do it.

**Reply by mladedav** [reply]
No, there's no way to filter out a span after it has been created.

If you want to make this decision when the span is closed you need the `Layer` implementation to do the filtering.

It should be possible to create something like `P99Layer<S, T: Layer<S>>(T, PhantomData<S>)` where you do some bookkeeping and if the timeout hasn't elapsed you just call the inner Layer.

Note though that this depends on the implementation of the inner layer. E.g. `tracing-opentelemetry` exports spans in `on_close` while other layers may export/print partial spans in `on_exit` which may be called multiple times, specifically both before the timeout and after the timeout for the same span.
